compiler/rustc_codegen_llvm/src/back/write.rs

```diff
@@ -576,12 +577,14 @@ pub(crate) unsafe fn llvm_optimize(
     // optimizations until after differentiation. Our pipeline is thus: (opt + enzyme), (full opt).
     // We therefore have two calls to llvm_optimize, if autodiff is used.
     if consider_ad && autodiff_stage != AutodiffStage::PostAD {
+        merge_functions = false;
```

Comment:
is there a way to do this on a per-function basis with some llvm attribute?

Reply:
Ok, after some experimenting I think it's not worth the cost here. Setting hasExternalLinkage will cause our body to be dropped (since it should be linked externally), so we drop/delete the function which we want to differentiate. That's a very effective way of preventing merging, but kind of problematic if we want to differentiate it. I could insert a random intrinsic and generate some arguments for it and add the Distinct metadata to one of the arguments, but I think the body of our dummy functions is already messy enough and I'd rather spend time towards moving to the declaration, which will sidestep this problem.